sources/SplaLibrary.cpp

```diff
@@ -41,6 +138,54 @@ void spla::Library::Submit(const spla::RefPtr<spla::Expression> &expression) {
     GetPrivate().GetExprManager()->Submit(expression);
 }
 
-class spla::LibraryPrivate& spla::Library::GetPrivate() {
+std::string spla::Library::PrintContextConfig() const noexcept {
+    std::stringstream confStream;
+    
+    const boost::compute::platform &platform = GetPrivate().GetPlatform();
+    const std::vector<boost::compute::device> &devices = GetPrivate().GetDevices();
+    confStream <<
+        "Platform: " << platform.name() << " (" << platform.id() << ")" << '\n' <<
+        "   Vendor: " << platform.vendor() << '\n' <<
+        "   Profile: " << platform.profile() << '\n' <<
+        "   Version: " << platform.version() << '\n' <<
+        "Total Devices: " << devices.size() << '\n';
+    for (const boost::compute::device &device : devices) {
+        confStream <<
+            "   * Device: " << device.name() << " (" << device.id()  << ")" << '\n' <<
+            "        Vendor: " << device.vendor() << '\n' <<
+            "        Profile: " << device.profile() << '\n' <<
+            "        Version: " << device.version() << '\n';
+    }
+    return confStream.str();
+}
+
+class spla::LibraryPrivate &spla::Library::GetPrivate() {
+    return *mPrivate;
+}
+
+const class spla::LibraryPrivate &spla::Library::GetPrivate() const noexcept {
     return *mPrivate;
 }
+
+spla::Library::Config::Config(const std::vector<std::string> &devices) : mDevicesNames(std::move(devices)) {}
+
+spla::Library::Config::Config() : Config({SelectDefaultDevice().name()}) {}
+
+spla::Library::Config::Config(
+    const std::string &platform,
+    DeviceType type,
+    DeviceAmount amount
+) : Config(SelectDevices(platform, type, amount)) {}
+
+spla::Library::Config::Config(
+    DeviceType type,
+    DeviceAmount amount
+) : Config(SelectDevices(type, amount)) {}
+
+spla::Library::Config::Config(
+    DeviceAmount amount
+) : Config(SelectDevices(amount)) {}
+
+const std::vector<std::string> &spla::Library::Config::GetDevicesNames() const noexcept {
+    return mDevicesNames;
+}
```

Comment:
Config class is simple storage with values, no actual selection, boost query and etc. must happen here.
This logic must be run only inside `LibraryPrivate` on `LibraryPrivate` instance creation.